### PR TITLE
OEUI-41: Patient Information and OpenMRS Navbar Integration

### PR DESCRIPTION
**Pull request by larrystone**
## **JIRA TICKET NAME:**
[OEUI-41: Patient Information and OpenMRS Navbar Integration](https://issues.openmrs.org/browse/OEUI-41)

### SUMMARY:
Integrate the current OpenMRS navigation bar and patient information into the order entry standalone OWA app.

### CODE REVIEW

File: app/js/routes.jsx [hunk 1]
```diff
@@ -6,17 +6,18 @@
...
     <Switch>
-      <Route path="/" component={App}></Route>
+      <Route path="/" component={OrderEntryPage} />
     </Switch>
-  );
-}
+  </div>
+);
+
+export default Routes;
```
Comment (fuchodeveloper) [inline app/js/routes.jsx]: Could you fix the eof error

Comment (larrystone) [reply app/js/routes.jsx]: Done, thanks for the feedback

**Comment by zeze111** [comment]
Looks good

File: app/js/actions/patient.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    .then((response) => {
+      dispatch({
+        type: SET_PATIENT,
+        patient: response.data,
+      });
+    })
+    .catch((error) => {
+      if (error.response) {
+        // TODO uncomment this to redirect user to homepage if uuid not present
+        // location.href = `/${applicationDistribution}`;
```
Comment (dkayiwa) [inline app/js/actions/patient.js]: When do you plan to address the above?

File: app/js/components/header/View.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                >
+                  {location.display}
+                </li>
+              ))}
+            </ul>
+          </div>
+        }
+      </li>
+      <li className="logout">
+        <a href="/openmrs-standalone/appui/header/logout.action?successUrl=openmrs-standalone">
```
Comment (dkayiwa) [inline app/js/components/header/View.jsx]: Are you sure everyone will use the standalone version?

File: app/js/actions/patient.js (hunk 2 above)
Comment (larrystone) [reply app/js/actions/patient.js]: This will be addressed when OEUI-40 is done, thanks

File: app/js/components/header/View.jsx (hunk 3 above)
Comment (larrystone) [reply app/js/components/header/View.jsx]: Oh my, the logic was to make the url dynamic, sorry I forgot this one, will fix it now

File: app/js/routes.jsx (hunk 1 above)
Comment (fuchodeveloper) [reply app/js/routes.jsx]: Awesome

File: app/js/components/header/View.jsx (hunk 3 above)
Comment (larrystone) [reply app/js/components/header/View.jsx]: Done, this has been fixed in a couple of places too

File: __mocks__/locationMock.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default {
+  location: {
+    href: 'http://localhost:3000/openmrs-standalone/owa/openmrs-owa-orderentry/index.html#',
```
Comment (dkayiwa) [inline __mocks__/locationMock.js]: Do you want the above to be standalone?

Comment (larrystone) [reply __mocks__/locationMock.js]: This is for test, it allows me to mock the url. This is necessary in cases where we run the automated tests on continuous integration service like travis where the window.location object is not available.

File: app/js/actions/header.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import axios from 'axios';
+import { SET_LOCATIONS, SET_CURRENT_SESSION } from './actionTypes';
+
+const applicationDistribution = window.location.href.split('/')[3];
+const apiBaseUrl = `/${applicationDistribution}/ws/rest/v1`;
+
+/**
+ * @description - fetch openMRS app locations
```
Comment (dkayiwa) [inline app/js/actions/header.js]: Regarding some of these comments, can you take a look at this? https://improvingsoftware.com/2011/06/27/5-best-practices-for-commenting-your-code/

File: tests/actions/patient.test.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+import mockData from '../../__mocks__/mockData';
+import locationMock from '../../__mocks__/locationMock';
+
+const middlewares = [thunk];
+
+const mockStore = configureMockStore(middlewares);
+window.location = locationMock;
+
+const applicationDistribution = window.location.href.split('/')[3];
```
Comment (dkayiwa) [inline tests/actions/patient.test.js]: Is the above the same as what you some times case appBaseUrl?

Comment (larrystone) [reply tests/actions/patient.test.js]: No, the apiBaseUrl was generated off the applicationDistribution
const apiBaseUrl = `/${applicationDistribution}/ws/rest/v1`;

File: app/js/actions/header.js (hunk 5 above)
Comment (larrystone) [reply app/js/actions/header.js]: Okay, I will look into

File: tests/actions/patient.test.js (hunk 6 above)
Comment (dkayiwa) [reply tests/actions/patient.test.js]: What does applicationDistribution mean? Can you give examples of possible values that it can take on?

Comment (larrystone) [reply tests/actions/patient.test.js]: Yes, applicationDistribution refers to the what the app core is running on.
`openmrs-standalone` could be a typical example

Comment (dkayiwa) [reply tests/actions/patient.test.js]: Wouldn't context path be a better name? Just google it a bit.

Comment (larrystone) [reply tests/actions/patient.test.js]: Thanks, will update accordingly

Comment (larrystone) [reply tests/actions/patient.test.js]: After some digging as suggested, I realised contextPath or applicationContext will do, so I choose contextPath, thanks for pointing this out

**Pull request merged**